packages/x-data-grid/src/utils/css/context.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useGridConfiguration } from '../../hooks/utils/useGridConfiguration';
+
+const CLASSNAME_PREFIX = 'MuiDataGridVariables';
+
+const CSSVariablesContext = React.createContext({
+  className: 'unset',
+  tag: <style href="/unset" />,
+});
+
+export function useCSSVariablesClass() {
+  return React.useContext(CSSVariablesContext).className;
+}
+
+export function useCSSVariables() {
+  return React.useContext(CSSVariablesContext);
+}
+
+export function GridPortalWrapper({ children }: { children: React.ReactNode }) {
+  const className = useCSSVariablesClass();
+  return <div className={className}>{children}</div>;
+}
+
+export function GridCSSVariablesContext(props: { children: any }) {
+  const config = useGridConfiguration();
+  const description = config.hooks.useCSSVariables();
+
+  const context = React.useMemo(() => {
+    const className = `${CLASSNAME_PREFIX}-${description.id}`;
+    const cssString = `.${className}{${variablesToString(description.variables)}}`;
+    const tag = <style href={`/${className}`}>{cssString}</style>;
```

Comment:
So the variables are scoped to a specific instance of the data grid 👍🏻
And then the className is applied to `GridRootStyles`.
Even though the `GridRootStyles` div cannot reach the CSS variables values (the `style` tag is a child of `GridRootStyles`), we can use CSS variables there because we only apply them to elements down the tree.
Do I get this correctly?

Reply:
It doesn't matter where the `<style>` tag is, it contains something like this:
```html
<style href="/DataGridVariables-123">
.DataGridVariables-123
  --DataGrid-t-spacing-unit: 4px;
  /* ... */
</style>
```
The `.DataGridVariables-123` classname is available globally even if the style tag is a child of the grid's root container.